src/api/make.js

```diff
@@ -52,9 +52,29 @@ export default async (providedOptions = {}) => {
     forgeConfig = await getForgeConfig(dir);
   });
 
-  if (platform && platform !== process.platform && !(process.platform === 'darwin' && platform === 'mas')) {
-    throw "You cannot run 'make' for a platform other than your system's platform";
-  }
+  const makers = {};
+  const targets = overrideTargets || forgeConfig.make_targets[platform];
+
+  targets.forEach((target) => {
+    const maker = requireSearchRaw(__dirname, [
+      `../makers/${platform}/${target}.js`,
+      `../makers/generic/${target}.js`,
+      `electron-forge-maker-${target}`,
+      target,
+      path.resolve(dir, target),
+      path.resolve(dir, 'node_modules', target),
+    ]);
+
+    if (!maker) {
+      throw `Could not find a build target with the name: ${target} for the platform: ${platform}`;
+    }
+
+    if (!maker.supportsPlatform) {
+      throw `Cannot build for ${platform} target ${target} from your ${process.platform} device`;
```

Comment:
Yeah, that's doable now that the makers are going to export a function instead of a constant. But is this really what we want to do? All the makers will have to update to be compatible with this version of `electron-forge` anyway because the maker's main entry point function signature changed, so why not have the authors update the supported platforms at the same time so all the makers are accurate, rather than some of them reporting that they support all platforms when they really don't? I'd think we'd want to prioritize users getting reliable information over maker maintainers having slightly more work in performing the update they'll already have to perform anyway.

I'll do whatever you prefer, but that's my two cents.